flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/legacy/metrics/MetricStore.java

```diff
@@ -538,19 +556,19 @@ public TaskMetricStore getTaskMetricStore(String taskID) {
     @ThreadSafe
     public static class TaskMetricStore extends ComponentMetricStore {
         private final Map<Integer, SubtaskMetricStore> subtasks;
-        private final Map<String, ComponentMetricStore> jmOperators;
+        private final ComponentMetricStore jmOperator;
```

Comment:
Can't there be multiple coordinators per task? A task can host multiple operators, each with their own coordinator. Is the idea to coalesce all of them into one metric store?

Reply:
After reading the rest of the code, this seems to be the case. Retrieving all the coordinator metrics at once for a task seems reasonable.